home-assignments/session1/solution2.py

```diff
@@ -0,0 +1,70 @@
+import requests
+import json
+
+
+# TODO: documantation
+# TODO: add try / except in requests part.
+# TODO: add try / except in writing file function
+
+
+def check_self_location_by_ip():
+    ip_api_url = "http://ip-api.com/json"
+    my_location = requests.get(url=ip_api_url)
+    my_location = my_location.json()
+    my_city = my_location['city']
+    my_country = my_location['country']
+    location = (my_city, my_country)
+
+    return location
+
+
+def get_weather_by_location(location):
+    city = location[0]
+    country = location[1]
+    weather_url = "http://api.openweathermap.org/data/2.5/weather?q={city},{country}&units=metric&APPID=07de467d22717f0eeee1c32550a7ebb1".format(
+        city=city, country=country)
+    weather = requests.get(url=weather_url)
+    weather = weather.json()
+
+    return weather
+
+
+def write_to_file(data, file_name):
+    with open(file_name, "w", encoding="utf-8") as f:
+        f.write(json.dumps(data))
+        exit(0)
```

Comment:
why exit(0) here? I think you need to add try/except statement and then add the exit(0)